pyomo/contrib/trustregion/Logger.py

```diff
@@ -8,8 +8,11 @@
 #  This software is distributed under the 3-clause BSD License.
 #  ___________________________________________________________________________
 
-import numpy as np
-from numpy.linalg import norm
+from pyomo.common.dependencies import numpy_available
+if numpy_available:
+    import numpy as np
+    from numpy.linalg import norm
```

Comment:
I would replace this with 
```python
from pyomo.common.dependencies import numpy as np
```
and then used `np.linalg.norm` explicitly in the module